### DISCUSSION TITLE
Building for iOS Gives Error

### DISCUSSION THREAD

**Discussion by ADLsourceCode** [flutter-stripe/flutter_stripe#878] [General]
`platform :ios, '13.0'
`

```
Flutter 2.10.3 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 7e9793dee1 (4 months ago) • 2022-03-02 11:23:12 -0600
Engine • revision bd539267b4
Tools • Dart 2.16.1 • DevTools 2.9.2

```

**Flutter Doctor**
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.10.3, on macOS 12.0.1 21A559 darwin-arm, locale
    en-AE)
[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2020.3)
[✓] VS Code (version 1.66.2)
[✓] Connected device (3 available)
[✓] HTTP Host Availability


```

Below is the error:-

```
Error output from Xcode build:
↳
    objc[14109]: Class AMSupportURLConnectionDelegate is implemented in both /usr/lib/libamsupport.dylib (0x1d9c77130) and /Library/Apple/System/Library/PrivateFrameworks/MobileDevice.framework/Versions/A/MobileDevice (0x1087202c8). One of the two will be used. Which one is undefined.
    objc[14109]: Class AMSupportURLSession is implemented in both /usr/lib/libamsupport.dylib (0x1d9c77180) and /Library/Apple/System/Library/PrivateFrameworks/MobileDevice.framework/Versions/A/MobileDevice (0x108720318). One of the two will be used. Which one is undefined.
    ** BUILD FAILED **


Xcode's output:
↳
    Writing result bundle at path:
    	/var/folders/xy/gyvq1qyj0kbbhw07knmgjtvh0000gn/T/flutter_tools.ivCN1a/flutter_ios_build_temp_dirMZcqZG/temporary_xcresult_bundle

    ld: file not found: /Users/akshaylamba/Library/Developer/Xcode/DerivedData/Runner-hglmusdhznnwzkhgsbvagtqnbpfn/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/StripeCore.build/Objects-normal/x86_64/StripeFile.o
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    ld: file not found: /Users/akshaylamba/Library/Developer/Xcode/DerivedData/Runner-hglmusdhznnwzkhgsbvagtqnbpfn/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/StripeCore.build/Objects-normal/arm64/StripeFile.o
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    error: the following command failed with exit code 1 but produced no further output
    Ld /Users/akshaylamba/Library/Developer/Xcode/DerivedData/Runner-hglmusdhznnwzkhgsbvagtqnbpfn/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/StripeCore.build/Objects-normal/arm64/Binary/StripeCore normal arm64 (in target 'StripeCore' from project 'Pods')
    note: Using new build system
    note: Planning
    note: Build preparation complete
    note: Building targets in dependency order
    /Users/akshaylamba/Documents/Pasiv/PasivFinal/pasiv-lite/ios/Pods/Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 15.2.99. (in target 'abseil' from project 'Pods')
    /Users/akshaylamba/Documents/Pasiv/PasivFinal/pasiv-lite/ios/Pods/Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 15.2.99. (in target 'gRPC-C++' from project 'Pods')
    /Users/akshaylamba/Documents/Pasiv/PasivFinal/pasiv-lite/ios/Pods/Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 15.2.99. (in target 'Toast' from project 'Pods')
    /Users/akshaylamba/Documents/Pasiv/PasivFinal/pasiv-lite/ios/Pods/Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 15.2.99. (in target 'FMDB' from project 'Pods')
    /Users/akshaylamba/Documents/Pasiv/PasivFinal/pasiv-lite/ios/Pods/Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 15.2.99. (in target 'leveldb-library' from project 'Pods')
    /Users/akshaylamba/Documents/Pasiv/PasivFinal/pasiv-lite/ios/Pods/Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 15.2.99. (in target 'gRPC-C++-gRPCCertificates-Cpp' from project 'Pods')
    /Users/akshaylamba/Documents/Pasiv/PasivFinal/pasiv-lite/ios/Pods/Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 15.2.99. (in target 'gRPC-Core' from project 'Pods')
    /Users/akshaylamba/Documents/Pasiv/PasivFinal/pasiv-lite/ios/Pods/Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 15.2.99. (in target 'BoringSSL-GRPC' from project 'Pods')

    Result bundle written to path:
    	/var/folders/xy/gyvq1qyj0kbbhw07knmgjtvh0000gn/T/flutter_tools.ivCN1a/flutter_ios_build_temp_dirMZcqZG/temporary_xcresult_bundle


Error (Xcode): File not found: /Users/akshaylamba/Library/Developer/Xcode/DerivedData/Runner-hglmusdhznnwzkhgsbvagtqnbpfn/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/StripeCore.build/Objects-normal/x86_64/StripeFile.o

Error (Xcode): File not found: /Users/akshaylamba/Library/Developer/Xcode/DerivedData/Runner-hglmusdhznnwzkhgsbvagtqnbpfn/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/StripeCore.build/Objects-normal/arm64/StripeFile.o

Could not build the application for the simulator.
Error launching application on iPhone 12 mini.

```

**Comment by remonh87** [comment]
It looks like there is something wrong in your caching. I would do `flutter clean` and `flutter pub get` first. Also it would be good to update xcode (it is still ok version but it wouldn't hurt)

**Comment by ADLsourceCode** [comment]
Still getting this error  @remonh87 

```

Xcode's output:
↳
    Writing result bundle at path:
    	/var/folders/xy/gyvq1qyj0kbbhw07knmgjtvh0000gn/T/flutter_tools.Qj5ZZR/flutter_ios_build_temp_dirbZX6aN/temporary_xcresult_bundle

    In file included from /Users/akshaylamba/Documents/Pasiv/PasivFinal/pasiv-lite/ios/Pods/gRPC-Core/src/core/lib/security/credentials/alts/alts_credentials.cc:21:
    In file included from /Users/akshaylamba/Documents/Pasiv/PasivFinal/pasiv-lite/ios/Pods/gRPC-Core/src/core/lib/security/credentials/alts/alts_credentials.h:27:
    /Users/akshaylamba/Documents/Pasiv/PasivFinal/pasiv-lite/ios/Pods/gRPC-Core/src/core/lib/security/credentials/credentials.h:205:15: warning: anonymous non-C-compatible type given name for linkage purposes by typedef declaration; add a tag name here [-Wnon-c-typedef-for-linkage]
    typedef struct {
                  ^
                   grpc_credentials_mdelem_array
    /Users/akshaylamba/Documents/Pasiv/PasivFinal/pasiv-lite/ios/Pods/gRPC-Core/src/core/lib/security/credentials/credentials.h:206:21: note: type is not C-compatible due to this default member initializer
      grpc_mdelem* md = nullptr;
                        ^~~~~~~
    /Users/akshaylamba/Documents/Pasiv/PasivFinal/pasiv-lite/ios/Pods/gRPC-Core/src/core/lib/security/credentials/credentials.h:208:3: note: type is given name 'grpc_credentials_mdelem_array' for linkage purposes by this typedef declaration
    } grpc_credentials_mdelem_array;
      ^
    1 warning generated.
    ld: file not found: /Users/akshaylamba/Library/Developer/Xcode/DerivedData/Runner-hglmusdhznnwzkhgsbvagtqnbpfn/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/StripeCore.build/Objects-normal/arm64/StripeFile.o
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    note: Using new build system
    note: Planning
    note: Build preparation complete
    note: Building targets in dependency order
    /Users/akshaylamba/Documents/Pasiv/PasivFinal/pasiv-lite/ios/Pods/Pods.xcodeproj: warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 15.2.99. (in target 'Toast' from project 'Pods')
    /Users/akshaylamba/Documents/Pasiv/PasivFinal/pasiv-lite/ios/Pods/Pods.xcodeproj: warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 15.2.99. (in target 'FMDB' from project 'Pods')
    /Users/akshaylamba/Documents/Pasiv/PasivFinal/pasiv-lite/ios/Pods/Pods.xcodeproj: warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 15.2.99. (in target 'BoringSSL-GRPC' from project 'Pods')
    /Users/akshaylamba/Documents/Pasiv/PasivFinal/pasiv-lite/ios/Pods/Pods.xcodeproj: warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 15.2.99. (in target 'abseil' from project 'Pods')
    /Users/akshaylamba/Documents/Pasiv/PasivFinal/pasiv-lite/ios/Pods/Pods.xcodeproj: warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 15.2.99. (in target 'gRPC-Core' from project 'Pods')
    warning: no rule to process file '/Users/akshaylamba/Documents/FlutterSDKs/2_10_3/.pub-cache/hosted/pub.dartlang.org/stripe_ios-3.3.0/ios/Classes/Stripe Sdk/StripeSdk.xcodeproj/project.pbxproj' of type 'text.pbxproject' for architecture 'arm64' (in target 'stripe_ios' from project 'Pods')
    warning: no rule to process file '/Users/akshaylamba/Documents/FlutterSDKs/2_10_3/.pub-cache/hosted/pub.dartlang.org/stripe_ios-3.3.0/ios/Classes/Stripe Sdk/StripeSdk.xcodeproj/xcshareddata/xcschemes/Tests.xcscheme' of type 'text.xml' for architecture 'arm64' (in target 'stripe_ios' from project 'Pods')
    /Users/akshaylamba/Documents/Pasiv/PasivFinal/pasiv-lite/ios/Pods/Pods.xcodeproj: warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 15.2.99. (in target 'gRPC-C++-gRPCCertificates-Cpp' from project 'Pods')
    /Users/akshaylamba/Documents/Pasiv/PasivFinal/pasiv-lite/ios/Pods/Pods.xcodeproj: warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 15.2.99. (in target 'leveldb-library' from project 'Pods')
    /Users/akshaylamba/Documents/Pasiv/PasivFinal/pasiv-lite/ios/Pods/Pods.xcodeproj: warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 15.2.99. (in target 'gRPC-C++' from project 'Pods')

    Result bundle written to path:
    	/var/folders/xy/gyvq1qyj0kbbhw07knmgjtvh0000gn/T/flutter_tools.Qj5ZZR/flutter_ios_build_temp_dirbZX6aN/temporary_xcresult_bundle


Could not build the precompiled application for the device.
Error (Xcode): File not found: /Users/akshaylamba/Library/Developer/Xcode/DerivedData/Runner-hglmusdhznnwzkhgsbvagtqnbpfn/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/StripeCore.build/Objects-normal/arm64/StripeFile.o


Error launching application on Akshay’s iPhone.

```

**Comment by remonh87** [comment]
can you try with latest version?

**Comment by Alfaizkhan** [comment]
@ADLsourceCode 
@remonh87 

Do this steps and try again.
- Add the latest version of the plugin in pubspec.yaml
`flutter_stripe: ^4.0.0`

- Delete the `pubspec.lock` as it contains the old data and run the `flutter clean && flutter pub get`

- in iOS folder, deleted the `pod.lock` and `pods` folder.

- Run the following command in iOS folder to regenerate the pods for **simulator**.
**`arch -x86_64 pod update`** 
- Run the following command in iOS folder to regenerate the pods for **physical device**.
**`pod install`**

- Run the project in simulator. 

After adding new dependancies or flutter updates,
This steps works for me every damn time. 😃

**Comment by remonh87** [comment]
Nice if this works for the @ADLsourceCode  I will update the troubleshoot section with this. 👍

**Comment by angus-gow** [comment]
Did this work? I'm having a very similar issue where I'm missing StripFile.o. I've followed all of the above and it's not working for me. I'm missing all the Stripe.o files and it's preventing me from building.

**Comment by DominicOrga** [comment]
I am also facing the same issue on my M1 machine.

**Comment by Alfaizkhan** [comment]
@DominicOrga 
using the same system(M1) and with above steps, working fine for me😸

**Comment by DominicOrga** [comment]
Hi @Alfaizkhan, thanks but unfortunately, same as @angus-gow that solution doesn't work for me and I'm still receiving the following issue:

```
Error (Xcode): File not found: /Users/dominicorga/Library/Developer/Xcode/DerivedData/Runner-ecrjhvggvywvowcbeeumzlyphppe/Build/Intermediates.noindex/ArchiveIntermediates/qa/IntermediateBuildFilesPath/Pods.build/Release-qa-iphoneos/StripeCore.build/Objects-normal/arm64/StripeFile.o
```

What I've also tried:
- Ensure that XCode is at latest stable release (Xcode 13.4.1)
- Clean the Flutter project (flutter clean && flutter pub get)
- Delete and re-generate pubspec.lock
- Delete and re-generate the pods
- Delete `flutter_stripe` from the `.pub_cache`
- Delete and re-generate `~/Library/Developer/Xcode/DerivedData/Runner-ecrjhvggvywvowcbeeumzlyphppe`

**Comment by remonh87** [comment]
@DominicOrga  can you go to the ios directory of your app and run `pod update` ?

**Comment by angus-gow** [comment]
I have done the same as @DominicOrga but went further in removing the entire .pub_cache running a "flutter pub cache clean" after doing a thorough clean using my own cleaning script:

#!/bin/zsh

flutter clean
cd ./packages/authentication_repository/
flutter clean
rm -f pubspec.lock
cd ../autoliv_user_repository/
flutter clean
rm -f pubspec.lock
cd ../../
echo "Flutter cleaned ==============================="
rm -Rf ~/Library/Developer/Xcode/DerivedData/*
echo "DerivedDate cleaned ==========================="
rm -Rf ~/Library/Caches/CocoaPods
echo "Cocoapods cleaned ============================="
rm -f ./pubspec.lock
echo "pubspec.lock deleted =========================="
echo "Cleanup IOS level components =================="
cd ./ios/
rm -Rf .symlinks
pod cache clean --all
echo "Pod Cache cleaned ============================="
pod deintegrate
echo "Pods deintegrated ============================="
rm ./Podfile.lock
echo "removed Podfile.lock =========================="
echo "Finished! ====================================="

I wiped the project locally and did a clean checkout from git.

This should take everything back to the beginning, I then put it back together, run it and get the same error. I'm of the opinion that there is something wrong with the config in Xcode so am now taking the nuclear option of uninstalling everything and creating a fresh install with new Xcode. 

The other developers in the team are not having this issue, so it's local to my setup for sure.

**Comment by DominicOrga** [comment]
Hi @remonh87, unfortunately, `pod update` doesn't work on my machine and I can only use `arch -x86_64 pod update`.
For more info on this issue, see https://github.com/CocoaPods/CocoaPods/issues/10220#issuecomment-730010626

Having said that, doing `arch -x86_64 pod update`, `arch -x86_64 pod install`, or `pod install` (this works for some reason) separately still leads to the same issue.

**Comment by remonh87** [comment]
Can you run `flutter doctor -v` and provide me the output?

**Comment by ms-biztech** [comment]
Thank you @remonh87 
Same issue in my system, i tried all the your steps 5 times but not working, Please help me to fix this issue.
My flutter doctor -v 

sys320new:example rem.ngr$ flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.0.5, on macOS 11.6 20G165 darwin-arm (Rosetta),
    locale en-IN)
[✓] Android toolchain - develop for Android devices (Android SDK version
    32.1.0-rc1)
[✓] Xcode - develop for iOS and macOS (Xcode 13.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.1)
[✓] VS Code (version 1.67.2)
Scanning for devices is taking a long time...[✓] Connected device (3 available)
    ! Error: M's iPad is not connected. Xcode will continue when Mau’s
      iPad is connected. (code -13)
[✓] HTTP Host Availability

• No issues found!

**Comment by remonh87** [comment]
@ms-biztech  can you give me the verbose output?  I would like to see your cocoapods version and compare it with me

**Comment by ms-biztech** [comment]
@remonh87 my Podfile.lock

PODS:
  - Flutter (1.0.0)
  - integration_test (0.0.1):
    - Flutter
  - pay_ios (0.0.1):
    - Flutter
  - Stripe (22.5.1):
    - Stripe/Stripe3DS2 (= 22.5.1)
    - StripeApplePay (= 22.5.1)
    - StripeCore (= 22.5.1)
    - StripeUICore (= 22.5.1)
  - Stripe/Stripe3DS2 (22.5.1):
    - StripeApplePay (= 22.5.1)
    - StripeCore (= 22.5.1)
    - StripeUICore (= 22.5.1)
  - stripe_ios (0.0.1):
    - Flutter
    - Stripe (~> 22.5.1)
    - StripeFinancialConnections (~> 22.5.1)
  - StripeApplePay (22.5.1):
    - StripeCore (= 22.5.1)
  - StripeCore (22.5.1)
  - StripeFinancialConnections (22.5.1):
    - StripeCore (= 22.5.1)
    - StripeUICore (= 22.5.1)
  - StripeUICore (22.5.1):
    - StripeCore (= 22.5.1)
  - webview_flutter_wkwebview (0.0.1):
    - Flutter

DEPENDENCIES:
  - Flutter (from `Flutter`)
  - integration_test (from `.symlinks/plugins/integration_test/ios`)
  - pay_ios (from `.symlinks/plugins/pay_ios/ios`)
  - stripe_ios (from `.symlinks/plugins/stripe_ios/ios`)
  - webview_flutter_wkwebview (from `.symlinks/plugins/webview_flutter_wkwebview/ios`)

SPEC REPOS:
  trunk:
    - Stripe
    - StripeApplePay
    - StripeCore
    - StripeFinancialConnections
    - StripeUICore

EXTERNAL SOURCES:
  Flutter:
    :path: Flutter
  integration_test:
    :path: ".symlinks/plugins/integration_test/ios"
  pay_ios:
    :path: ".symlinks/plugins/pay_ios/ios"
  stripe_ios:
    :path: ".symlinks/plugins/stripe_ios/ios"
  webview_flutter_wkwebview:
    :path: ".symlinks/plugins/webview_flutter_wkwebview/ios"

SPEC CHECKSUMS:
  Flutter: 50d75fe2f02b26cc09d224853bb45737f8b3214a
  integration_test: a1e7d09bd98eca2fc37aefd79d4f41ad37bdbbe5
  pay_ios: 8c7beb9c61d885f3f51b61f75f8793023fc8843a
  Stripe: 2b2decd03146e08a350960966d41f3028c2eac29
  stripe_ios: e94186250816de030881ced3399154eab186e439
  StripeApplePay: b14f06ac6fc24b56704c1e598149ed0cc45e166f
  StripeCore: 4833738f2ca4336712f279f3c2867a0a7eb67c93
  StripeFinancialConnections: 982115b82af429968d8aa78d329a42ed7ba3feab
  StripeUICore: 08c1efbd7e3c54ee7fa74334a37a1d4c08ba944d
  webview_flutter_wkwebview: b7e70ef1ddded7e69c796c7390ee74180182971f

PODFILE CHECKSUM: 7be2f5f74864d463a8ad433546ed1de7e0f29aef

COCOAPODS: 1.11.3

**Comment by DominicOrga** [comment]
Here's my the output for `flutter doctor -v`:

```
[✓] Flutter (Channel stable, 3.0.5, on macOS 12.5 21G72 darwin-arm, locale en-PH)
    • Flutter version 3.0.5 at /Users/dominicorga/fvm/versions/3.0.5
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f1875d570e (4 weeks ago), 2022-07-13 11:24:16 -0700
    • Engine revision e85ea0e79c
    • Dart version 2.17.6
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/dominicorga/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)

[✓] VS Code (version 1.70.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.46.0

[✓] Connected device (3 available)
    • iPhone 13 mini (mobile) • 00008110-00046C3411B8801E • ios            • iOS 15.6 19G71
    • macOS (desktop)         • macos                     • darwin-arm64   • macOS 12.5 21G72 darwin-arm
    • Chrome (web)            • chrome                    • web-javascript • Google Chrome 104.0.5112.79
    ! Error: iPhone 13 mini is busy: Preparing the watch for development via iPhone 13 mini. Xcode will continue when iPhone 13 mini is finished. (code
      -10)

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

**Comment by ms-biztech** [comment]
@remonh87 my lutter doctor --verbose

/Users/remila.nagar/Installer/flutter/bin/flutter doctor --verbose
[✓] Flutter (Channel stable, 3.0.5, on macOS 11.6 20G165 darwin-arm, locale en-IN)
    • Flutter version 3.0.5 at /Users/remila.nagar/Installer/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f1875d570e (4 weeks ago), 2022-07-13 11:24:16 -0700
    • Engine revision e85ea0e79c
    • Dart version 2.17.6
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at /Users/remila.nagar/Library/Android/sdk
    • Platform android-32, build-tools 32.1.0-rc1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)

[✓] VS Code (version 1.67.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (3 available)
    • iPhone 13 mini (mobile) • EF61C145-9C71-4C89-ADE2-18E238061355 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-15-0 (simulator)
    • macOS (desktop)         • macos                                • darwin-arm64   • macOS 11.6 20G165 darwin-arm
    • Chrome (web)            • chrome                               • web-javascript • Google Chrome 104.0.5112.79
    ! Error: Maulik’s iPad is not connected. Xcode will continue when Maulik’s iPad is connected. (code -13)

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
Process finished with exit code 0

**Comment by ms-biztech** [comment]
@remonh87 I think my Xcode version is to old 13.1

**Comment by DominicOrga** [comment]
There's definitely something weird going on with the `Stripe.o` file being generated.

When cleaning, re-installing pods and running/building the app, two possible errors may occur, either:

```
Error (Xcode): Can't open file, errno=1 file '/Users/dominicorga/Library/Developer/Xcode/DerivedData/Runner-ecrjhvggvywvowcbeeumzlyphppe/Build/Intermediates.noindex/Pods.build/Debug-dev-iphoneos/StripeCore.build/Objects-normal/arm64/StripeFile.o'
```
or 
```
Error (Xcode): File not found: /Users/dominicorga/Library/Developer/Xcode/DerivedData/Runner-ecrjhvggvywvowcbeeumzlyphppe/Build/Intermediates.noindex/ArchiveIntermediates/qa/IntermediateBuildFilesPath/Pods.build/Release-qa-iphoneos/StripeCore.build/Objects-normal/arm64/StripeFile.o
```

When you try navigating to the `StripeFile.o` directory, the file may or may not be there. If its there, then after some time it suddenly disappears.

<img width="919" alt="Screen Shot 2022-08-12 at 9 08 20 PM" src="https://user-images.githubusercontent.com/12520299/184360453-0fac27b6-cbe0-4868-8910-af3abb74b884.png">

In addition, I tried opening the `StripeFile.o` but it failed, then it suddenly disappeared. I tried doing it on the other .o files and they seem to work fine.

**Comment by ms-biztech** [comment]
@remonh87 What is next??

**Comment by remonh87** [comment]
I would recommend to try to build your app from xcode itself insteald of the flutter tooling. Sometimes this helps. Else I would recommend creating an example repository for us so we can try to reproduce this issue.  Since both my professional projects and the example app compile without issues.

**Comment by DominicOrga** [comment]
Just tried creating a new app with `flutter_stripe` and it still returns the same result regardless when built from Flutter or XCode.

 The app works fine when the `flutter_stripe` package is removed.

**Steps to re-create on my Mac M1:**
1. Create a new flutter project (`flutter create stripe_test`)
2. Add `flutter_stripe 4.0.0` in pubspec.yaml
3. `flutter clean && flutter pub get`
4. Update IOS Platform version in PodFile (`platform :ios, '13.0'`)
5. Install pod dependencies (`arch -x86_64 pod install`)
6. Run in XCode

**Comment by DominicOrga** [comment]
Other things I've tried that doesn't work:
- Reinstall XCode
- Reinstall cocoapods

**Comment by ms-biztech** [comment]
Thank you! @remonh87 I'm not getting this issue in Xcode 13.4.1 with the latest stripe_flutter 4.0.0 plugin in my another mac system. Working fine, May be this issue is happening in the the old Xcode version. Thanks again @remonh87

**Comment by DominicOrga** [comment]
@ms-biztech is your other mac system intel or m1?

**Comment by ms-biztech** [comment]
@DominicOrga My mac system is m1.

**Comment by remonh87** [comment]
@DominicOrga  I also have an m1 lets convert this to a discussion now so we can help you and provide support. IT is something in your config that is wrong but doesn't seem to be in the library.

**Comment by DominicOrga** [comment]
@remonh87 I agree this seems to be an issue on my machine. We tried running it on another m1 machine and it successfully builds. I tried clearing XCode's cache, reinstalling XCode, and building the app directly from XCode but to no avail.

Here is a sample Flutter project that I use for testing:
https://github.com/DominicOrga/stripe_test

To run the app, you can call `tool/build.sh`.

**Comment by DominicOrga** [comment]
For more context, here's a screenshot of all the errors displayed in XCode.

<img width="1253" alt="Screen Shot 2022-08-16 at 1 32 03 PM" src="https://user-images.githubusercontent.com/12520299/184804968-cf09b6d2-e593-4d17-9f1a-af9a511e8e1b.png">
<img width="1253" alt="Screen Shot 2022-08-16 at 1 32 12 PM" src="https://user-images.githubusercontent.com/12520299/184804974-709f85cb-eb63-4838-9366-ec563c31dcd9.png">
<img width="1249" alt="Screen Shot 2022-08-16 at 1 32 22 PM" src="https://user-images.githubusercontent.com/12520299/184804979-03308ba7-a97e-48e1-bb2f-0debff0d885d.png">

**Comment by daveols** [comment]
After struggling with this exact same issue for more time than I care to admit, I found out the cause of the issue on my local machine.

We're using an endpoint protection tool across our device fleet and it turns out the `Stripe.o` build file was being quarantined as soon as it was created leading to the build failing locally. This didn't happen on version `^2.5.0` of this library but something in the newer versions of the Stripe binary is causing this false positive to occur. Until now we were blocked from upgrading to Flutter 3+ due to this library failing to build.

We were able to resolve this by adding a real-time detection exclusion rule for this specific file in the configuration of the endpoint security tool.

If you're running a similar setup, try checking the logs of your tool or disabling it altogether to see if you have the same issue.

Thank you @DominicOrga for nudging us in the right direction with your missing file analysis.

@remonh87 – any idea why this newer version might be triggering malware protection?

**Reply by lohitbura07** [reply]
Can you please tell exactly what I should do to resolve it ?

**Comment by lohitbura07** [comment]
Facing the Same Issue . Any Conclusion to this Conversation ?

Flutter - 3.3.0
Xcode - 14
Stripe - 5.0.0

**Comment by Fakher-Hakim** [comment]
same issue here !
